test/powershell/Modules/Microsoft.PowerShell.Utility/ConvertFrom-Json.Tests.ps1

```diff
@@ -52,4 +75,84 @@ Describe 'ConvertFrom-Json' -tags "CI" {
             $json | Should -BeOfType Hashtable
         }
     }
+
+    It 'Can convert an object of depth 1024 by default with AsHashtable switch set to <AsHashtable>' -TestCases $testCasesWithAndWithoutAsHashtableSwitch {
+        Param($AsHashtable)
+        $nestedJson = GenerateNestedJson -Depth:1024
+
+        $json = $nestedJson | ConvertFrom-Json -AsHashtable:$AsHashtable
+
+        if ($AsHashtable)
+        {
+            $json | Should -BeOfType Hashtable
+        }
+        else
+        {
+            $json | Should -BeOfType PSCustomObject
+        }
+    }
+
+    It 'Fails to convert an object of depth higher than 1024 by default with AsHashtable switch set to <AsHashtable>' -TestCases $testCasesWithAndWithoutAsHashtableSwitch {
+        Param($AsHashtable)
+        $nestedJson = GenerateNestedJson -Depth:1989
+
+        { $nestedJson | ConvertFrom-Json -AsHashtable:$AsHashtable } |
+            Should -Throw -ErrorId "System.ArgumentException,Microsoft.PowerShell.Commands.ConvertFromJsonCommand"
+    }
+
+    It 'Can convert an object with depth less than Depth param set to <Depth> and AsHashtable switch set to <AsHashtable>' -TestCases $testCasesJsonDepthWithAndWithoutAsHashtableSwitch {
+        Param($AsHashtable, $Depth)
+        $nestedJson = GenerateNestedJson -Depth:($Depth - 1)
+
+        $json = $nestedJson | ConvertFrom-Json -AsHashtable:$AsHashtable -Depth:$Depth
+
+        if ($AsHashtable)
+        {
+            $json | Should -BeOfType Hashtable
```

Comment:
does it make sense to verify the result has the correct depth?

Reply:
it does! this makes the tests take a bit more time to complete, so definitely moving them to feature tests.